django/db/backends/oracle/features.py

```diff
@@ -139,6 +139,25 @@ def django_test_skips(self):
                     },
                 }
             )
+        if self.connection.is_pool:
```

Comment:
Some tests hang for me when running with pool as True, such as:
- `select_for_update.tests.SelectForUpdateTests.test_raw_lock_not_available`
- `select_for_update.tests.SelectForUpdateTests.test_nowait_raises_error_on_block`
- `backends.tests.ThreadTests.test_default_connection_thread_local`

Can you check why?

Reply:
After debugging the test `select_for_update.tests.SelectForUpdateTests.test_raw_lock_not_available`, I found that the test requires three connections. It attempts to acquire three connections without releasing them back to the pool. Since you have set `pool: True`, it uses the driver's default values, which set the maximum pool size to 2. Consequently, the pool becomes full, and the third acquire request hangs. Same for the other two tests.

While running the test suite, you can adjust the configuration to `'OPTIONS': {'pool': {'max': 3}}`. Alternatively, you could increase the maximum value even further, to 5 or 10. This change will fix the problem.

Reply:
@sarahboyce The other option is to set the "get" mode to FORCE which will create new connections anyway when the pool is full. So you can the configuration as `'OPTIONS':{'pool':{'getmode':oracledb.POOL_GETMODE_FORCEGET}},` . With this option even if max is 2 new connection will be created if required. Although I believe 1st will be the best option.

Reply:
I see, thank you!
These are different defaults to the [postgres connection pooling](https://www.psycopg.org/psycopg3/docs/api/pool.html#psycopg_pool.ConnectionPool) (which has a timeout for example) but that's fine
I like that we don't have an "opinion" as to sensible defaults and don't repeat the documentation of the external library. So in short, I think this requires no change

Reply:
Yes, the default values for PostgreSQL connection pooling parameters differ. We discussed what the defaults should be and decided to leave them up to the driver. This approach helps keep the documentation and code simpler and more aligned with any changes in PostgreSQL. You can view the discussion here https://github.com/django/django/pull/17834#discussion_r1738024240.
Thank You.